src/components/cylc/analysis/TimeSeries.vue

```diff
@@ -0,0 +1,459 @@
+          },
+          labels: {
+            formatter: function (value) {
+              return formatDuration(value, true)
+            }
+          },
+        },
+      }
+    },
+
+    miniChartOptions () {
+      return {
+        chart: {
+          animations: {
+            enabled: this.animate && !this.reducedAnimation,
+            easing: 'easeinout',
+            speed: 300,
+            animateGradually: {
+              enabled: true,
+              delay: 150,
+            },
+            dynamicAnimation: {
+              enabled: true,
+              speed: 350,
+            },
+          },
+          selection: {
+            enabled: true,
+            xaxis: {
+              min: 1,
+              max: this.cyclePoints.length,
+            }
+          },
+          toolbar: {
+            autoSelected: 'selection',
+            show: false
+          }
+        },
+        legend: {
+          show: false
+        },
+        markers: {
+          size: 3
+        },
+        stroke: {
+          width: 2
+        },
+        tooltip: {
+          enabled: false
+        },
+        xaxis: {
+          categories: this.cyclePoints,
+          tickAmount: 4,
+          labels: {
+            rotate: 0
+          }
+        },
+        yaxis: {
+          tickAmount: 3,
+          title: {
+            text: upperFirst(this.timingOption) + ' time',
+          },
+          labels: {
+            formatter: function (value) {
+              return formatDuration(value, true)
+            }
+          },
+          min: this.showOrigin ? 0 : undefined
+        },
+      }
+    },
+  },
+
+  methods: {
+    selectSearchResults: function () {
+      // Do we need a limit to number of tasks that can be added?
+      const filteredTasks = this.$refs.selectTasks.filteredItems.map(
+        (filteredTask) => (filteredTask.value))
+      this.displayedTasks = union(this.displayedTasks, filteredTasks)
+    },
+    deselectSearchResults: function () {
+      const filteredTasks = this.$refs.selectTasks.filteredItems.map(
+        (filteredTask) => (filteredTask.value))
+      this.displayedTasks = difference(this.displayedTasks, filteredTasks)
+    },
+    jobsQuery: debounce(
+      async function (queryTasks) {
+        // Ensure query isn't run over all tasks
+        if (queryTasks.length > 0) {
+          this.jobs = []
+          this.jobCallback = new AnalysisJobCallback(this.jobs)
+          const retJob = await this.$workflowService.query2(
+            JOB_QUERY,
+            { workflows: this.workflowIDs, tasks: queryTasks }
+          )
+          this.jobCallback.onAdded(retJob.data)
+        }
+      },
+      200 // only re-run this once every 0.2 seconds
+    ),
+    zoomMainChart: function (context, { xaxis }) {
+      this.xRange = [Math.ceil(xaxis.min), Math.floor(xaxis.max)]
+    }
+  },
+
+  icons: {
+    mdiRefresh,
+  },
+}
+</script>
+
+<style>
+.apexcharts-text {
+  font-size: 0.9rem;
+}
+</style>
```

Comment:
I think we might as well pull this out into [`index.scss`](https://github.com/cylc/cylc-ui/blob/63672fd32ccc2465b627850fdb0d97e079f3b34a/src/styles/index.scss) seeing as it's being applied to all the apexcharts